src/flow-worker.c

```diff
@@ -620,7 +620,9 @@ static TmEcode FlowWorker(ThreadVars *tv, Packet *p, void *data)
                 StreamTcpSessionCleanup(p->flow->protoctx);
             }
         } else if (p->proto == IPPROTO_TCP && p->flow->protoctx) {
-            FramesPrune(p->flow, p);
+            if (p->flags & PKT_STREAM_EST) {
```

Comment:
should we still call `StreamTcpPruneSession` in this case? Seems not. Why would an error packet have any effect on state?

Reply:
ok